### PR TITLE
Issue #2453533: provided interface for RulesState

### PR DESCRIPTION
**Pull request by p4trizio**
rebased

### CODE REVIEW

File: src/Engine/RulesStateInterface.php [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+use Drupal\Core\Language\LanguageInterface;
+use Drupal\Core\Plugin\Context\ContextInterface;
+use Drupal\rules\Exception\RulesEvaluationException;
+
+/**
+ * Defines the interface for Rules states.
+ *
+ */
+interface RulesStateInterface {
```
Comment (fago) [inline src/Engine/RulesStateInterface.php:18]: Interface is not used by the implementing class.

File: src/Engine/RulesStateInterface.php [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+namespace Drupal\rules\Engine;
+
+use Drupal\Core\Language\LanguageInterface;
+use Drupal\Core\Plugin\Context\ContextInterface;
+use Drupal\rules\Exception\RulesEvaluationException;
+
+/**
+ * Defines the interface for Rules states.
+ *
```
Comment (fago) [inline src/Engine/RulesStateInterface.php:16]: One new line too much.

File: src/Engine/RulesStateInterface.php [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+ */
+
+namespace Drupal\rules\Engine;
+
+use Drupal\Core\Language\LanguageInterface;
+use Drupal\Core\Plugin\Context\ContextInterface;
+use Drupal\rules\Exception\RulesEvaluationException;
+
+/**
+ * Defines the interface for Rules states.
```
Comment (fago) [inline src/Engine/RulesStateInterface.php:15]: Defines an interface or the rules state.

**Comment by klausi** [comment]
Continued in #168

**Pull request closed without merging**